Bundled dependencies built with browserify broken

The newest version (3.1.0) broke some of my dependencies.

For example, I bundle video.js which is built with `grunt-browserify`. The first line of their [`dist/video.js`](https://github.com/videojs/video.js/raw/v5.10.6/dist/video.js) has a [`umd`](https://www.npmjs.com/package/umd) wrapper around browserify's [`browser-pack`](https://www.npmjs.com/package/browser-pack) prelude:
```js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.videojs = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(_dereq_,module,exports){
```

There are two problematic lines in the `browser-pack` prelude:
* `var a=typeof require=="function"&&require;`
* `var i=typeof require=="function"&&require;`

Version 3.1.0 of this plugin transforms these lines into:
* `var a='function'=="function"&&require;`
* `var i='function'=="function"&&require;`

This breaks the code at run time inside a browser with strict mode enabled, since there's no global `require` function.

For now, I've pinned my version of `rollup-plugin-commonjs` to `=3.0.2`, so that my builds keep working.

I have not yet found the time to set up a full reproduction case, but this is the general idea:

1. Create `foo.js` and bundle it with browserify. (In practice, we'd get the bundled `foo.js` from an npm dependency.)
2. Create `bar.js` that `import`s from the built `foo.js` and bundle it with rollup.
3. Load `bar.js` inside a browser.

--- a/src/helpers.js
+++ b/src/helpers.js
@@ -3,6 +3,10 @@ export const HELPERS_ID = '\0commonjsHelpers';
 export const HELPERS = `
 export var commonjsGlobal = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {};
 
+export function commonjsRequire () {
+	throw new Error('Dynamic requires are not currently supported by rollup-plugin-commonjs');
+}
+
 export function unwrapExports (x) {
 	return x && x.__esModule ? x['default'] : x;
 }

--- a/src/transform.js
+++ b/src/transform.js
@@ -59,7 +59,7 @@ export default function transform ( code, id, isEntry, ignoreGlobal, customNamed
 	let uid = 0;
 
 	let scope = attachScopes( ast, 'scope' );
-	const uses = { module: false, exports: false, global: false };
+	const uses = { module: false, exports: false, global: false, require: false };
 
 	let lexicalDepth = 0;
 	let programDepth = 0;
@@ -126,22 +126,16 @@ export default function transform ( code, id, isEntry, ignoreGlobal, customNamed
 				return;
 			}
 
-			// To allow consumption of UMD modules, transform `typeof require` to `'function'`
-			if ( node.type === 'UnaryExpression' && node.operator === 'typeof' && node.argument.type === 'Identifier' ) {
-				const name = node.argument.name;
-
-				if ( name === 'require' && !scope.contains( name ) ) {
-					magicString.overwrite( node.start, node.end, `'function'` );
-					return;
-				}
-			}
-
 			if ( node.type === 'Identifier' ) {
 				if ( isReference( node, parent ) && !scope.contains( node.name ) ) {
 					if ( node.name in uses ) {
 						uses[ node.name ] = true;
 						if ( node.name === 'global' && !ignoreGlobal ) {
 							magicString.overwrite( node.start, node.end, `${HELPERS_NAME}.commonjsGlobal`, true );
+						}
+
+						if ( node.name === 'require' ) {
+							magicString.overwrite( node.start, node.end, `${HELPERS_NAME}.commonjsRequire`, true );
 						}
 
 						// if module or exports are used outside the context of an assignment
@@ -189,6 +183,8 @@ export default function transform ( code, id, isEntry, ignoreGlobal, customNamed
 				// is a bare import, e.g. `require('foo');`
 				magicString.remove( parent.start, parent.end );
 			}
+
+			node.callee._skip = true;
 		},
 
 		leave ( node ) {

--- a/test/function/typeof-require/foo.js
+++ b/test/function/typeof-require/foo.js
@@ -1,0 +1,5 @@
+if ( typeof require === 'function' && require ) {
+	module.exports = 1;
+} else {
+	module.exports = 2;
+}

--- a/test/function/typeof-require/main.js
+++ b/test/function/typeof-require/main.js
@@ -1,0 +1,3 @@
+import foo from './foo.js';
+
+assert.equal( foo, 1 );

